components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/apikey/ApiKeyAuthenticator.java

```diff
@@ -358,6 +361,71 @@ public AuthenticationResponse authenticate(MessageContext synCtx) {
         }
     }
 
+    private void validateAPIKeyRestrictions(JWTClaimsSet payload, MessageContext synCtx) throws APISecurityException {
+        org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) synCtx).
+                getAxis2MessageContext();
+
+        String permittedIPList = (String) payload.getClaim(APIConstants.JwtTokenConstants.PERMITTED_IP);
```

Comment:
null check claim not available at none state

Reply:
null can be type cast to String and it still will be null. So, do we need to check for null here?

Reply:
fixed